README.md

```diff
@@ -2,6 +2,160 @@
+    # cluster: greenfield
+  env:
+    SOME_API_KEY: ${{ secrets.SOME_API_KEY }}
+```
+
+This GitHub action takes the following inputs:
+
+| Name      | Description |
+|-----------|-------------|
+| `path`    | _(Required)_ The path to a directory containing transformation manifests. |
+| `api-key` | _(Required)_ The API key used for authenticating with transformations. Equivalent to setting the `JETFIRE_API_KEY` environment variable. |
+| `cluster` | _(Optional)_ The name of the cluster where Jetfire/Transformations is hosted. |
+
+Additionally, you must specify environment variables for any API keys referenced in transformation manifests.
+
+
+See [workflow.example.yml](workflow.example.yml) for a complete example.
+
+#### Manifest file format
+
+```yaml
+# Required
+externalId: my-transformation
+
+
+# Required
+name: My Transformation
+
+
+# Optional, default: false
+shared: true
+
+
+# Optional, default: null
+# If null, transformation will not be scheduled.
+schedule: "1 * * * *"
+
+
+# Required
+# The path to a file containing the SQL query for this transformation.
+query: my-transformation.sql
+
+
+# Required
+# Valid values are: assets, assethierarchy, events, timeseries, datapoints, stringdatapoints
+destination: datapoints
+# When writing to RAW tables, use the following syntax:
+destination:
+  type: raw
+  rawDatabase: some_database
+  rawTable: some_table
+
+
+# Optional, default: upsert
+# Valid values are:
+#   upsert: Create new items, or update existing items if their id or externalId already exists.
+#   create: Create new items. Transformation will fail if there are id or externalId conflicts.
+#   update: Update existing items. Transformation will fail if an id or externalId does not exist.
+#   delete: Delete items by internal id.
+action: update
+
+
+# Required
+# The API key that will be used to run the transformation,
+# specified as the name of an environment variable.
+apiKey: SOME_ENVIRONMENT_VARIABLE
+# If you want to use separate API keys for reading/writing data, use the following syntax:
+apiKey:
+  read: READ_API_KEY
+  write: WRITE_API_KEY
+```
+
+
+### Make a query
+
+`jetfire-cli` also allows you to run queries, and retrieve up to 1000 rows of results.
+
+```sh
+jetfire query "select * from _cdf.assets limit 100"
+```
+
+This will print a JSON document to standard output containing the results of your query.
+
+#### Query limits
+
+The query command is intended for previewing your SQL queries, and is not designed to produce bigger data dumps.
+For this reason, there are a few limits in place.
+
+First of all, there is a final limit of 1000 rows that can be returned from your query.
+This is a limitation in the backend itself, although this may change in the future.
+This is equivalent to a final LIMIT clause on your query - aggregations and joins may still process more than 1000 rows.
+
+There is also a _source limit_, which is configurable using the `--source-limit` parameter.
```

Comment:
Are we always using one partition then? Otherwise this is a bit of a lie.

Reply:
Yes, for queries, we always set `partitions = Some(1)`.